src/annotator/anchoring/test/types-test.js

```diff
@@ -19,43 +21,290 @@ describe('types', function () {
     document.body.appendChild(container);
   });
 
-  after(function () {
+  afterEach(() => {
     container.remove();
   });
 
-  describe('TextQuoteAnchor', function () {
-    describe('#toRange', function () {
-      it('returns a valid DOM Range', function () {
+  describe('RangeAnchor', () => {
+    let fakeSniff;
+    let fakeSerializedRange;
+    let fakeNormalize;
+    let fakeSerialize;
+
+    beforeEach(() => {
+      fakeSerializedRange = sinon.stub();
+      fakeSerialize = sinon.stub();
+      fakeNormalize = sinon.stub().returns({
+        serialize: fakeSerialize,
+      });
+      fakeSniff = sinon.stub().returns({
+        normalize: fakeNormalize,
+      });
+      $imports.$mock({
+        './range': {
+          sniff: fakeSniff,
+          SerializedRange: fakeSerializedRange,
+        },
+      });
+    });
+
+    afterEach(() => {
+      $imports.$restore();
+    });
+    function createRangeAnchor() {
+      return new Range();
+    }
+
+    describe('#constructor', () => {
+      it('throws an error if `root` is missing', () => {
+        assert.throws(() => {
+          new RangeAnchor();
+        }, 'missing required parameter "root"');
+      });
+
+      it('throws an error if `range` is missing', () => {
+        assert.throws(() => {
+          new RangeAnchor({});
+        }, 'missing required parameter "range"');
+      });
+    });
+
+    describe('#fromRange', () => {
+      it('returns a new RangeAnchor instance', () => {
+        const anchor = RangeAnchor.fromRange(container, createRangeAnchor());
+        assert.calledOnce(fakeNormalize);
```

Comment:
I think it would be better to test that `anchor.range` is the result of whatever `fakeNormalize` returns. This check just verifies that `fakeNormalize` was called but that doesn't mean that `RangeAnchor`'s anchor is actually a normalized range.